### PR TITLE
Implement second pipe function to 'download' from nextcloud

### PR DESCRIPTION
**Pull request by kevinley**
- added function to pipe data located at the provided path into the provided writable stream
- renamed `pipeStream` to `uploadFromStream` and `downloadToStream`

- [x] I have carefully evaluated the risks of this change
- [x] This change does not remove existing functionality
- [x] I have written tests to cover all risks
- [x] I updated the docs

### CODE REVIEW

File: source/webdav.ts [hunk 1]
```diff
@@ -187,31 +187,33 @@ export async function checkConnectivity(): Promise<boolean> {
...
-export const touchFolder           = clientFunction(rawTouchFolder);
-export const pipeStream            = clientFunction(rawPipeStream);
+export const createFolderHierarchy   = clientFunction(rawCreateFolderHierarchy);
+export const getFolderFileDetails    = clientFunction(rawGetFolderFileDetails);
+export const getFolderProperties     = clientFunction(rawGetFolderProperties);
+export const getWriteStream          = clientFunction(rawGetWriteStream);
+export const getReadStream           = clientFunction(rawGetReadStream);
+export const touchFolder             = clientFunction(rawTouchFolder);
+
+export const pipeStream              = clientFunction(rawUpload); // deprecated
```
Comment (kwisatz) [inline source/webdav.ts]: We should probably emit a warning?! How would one do that in a library ?

File: tests/webdav-jest.ts [hunk 2]
```diff
@@ -421,30 +423,50 @@ describe('Webdav integration', function testWebdavIntegration() {
     });
   });
 
-  describe('pipeStream(path, stream)', () => {
-    it('should pipe readable streams to the Nextcloud instance', async () => {
+  describe('upload(targetPath, readStream)', () => {
+    it('should pipe data to the Nextcloud instance', async () => {
```
Comment (kwisatz) [inline tests/webdav-jest.ts]: Well, it's not wrong that `upload` and `download` both still expect streams, is it?

File: tests/webdav-jest.ts [hunk 3]
```diff
@@ -421,30 +423,50 @@ describe('Webdav integration', function testWebdavIntegration() {
...
-      await client.pipeStream(path, stream);
+      await client.upload(path, readStream);
 
       expect(await client.get(path)).toBe(string);
 
       await client.remove(path);
     });
   });
 
+  fdescribe('download(sourcePath, writeStream)', () => {
```
Comment (kwisatz) [inline tests/webdav-jest.ts]: `fdescribe` ?

File: source/types.ts [hunk 4]
```diff
@@ -30,7 +30,9 @@ export interface NextcloudClientInterface extends NextcloudClientProperties {
   getFolderProperties(path: string, extraProperties?: FileDetailProperty[]):  Promise<FolderProperties>;
   configureWebdavConnection(options: ConnectionOptions):                      void;
   configureOcsConnection(options: ConnectionOptions):                         void;
-  pipeStream(path: string, stream: Stream.Readable):                          Promise<void>;
+  pipeStream(path: string, readStream: Stream.Readable):                      Promise<void>;
+  upload(targetPath: string, readStream: Stream.Readable):                    Promise<void>;
```
Comment (kwisatz) [inline source/types.ts]: What about `uploadFromStream` and `downloadToStream` ?
Or, to stay with `webdav` terms:

`postFromStream` and `getAsStream`

File: source/webdav.ts (hunk 1 above)
Comment (kwisatz) [reply source/webdav.ts]: https://github.com/nodejs/node/issues/11642
https://github.com/dougwilson/nodejs-depd

Comment (kevinley) [reply source/webdav.ts]: Thought the same. Will do process.emitWarning

File: source/types.ts (hunk 4 above)
Comment (kevinley) [reply source/types.ts]: uploadFromStream and downloadToStream sound ok

File: tests/webdav-jest.ts (hunk 3 above)
Comment (kevinley) [reply tests/webdav-jest.ts]: thanks, forgot to remove the focus

**Comment by floriansimon1** [comment]
Do you think just removing the function would be better? Sure, that requires amajor version bump but it might be worth it given the small adoption the library has

**Comment by kwisatz** [comment]
> Do you think just removing the function would be better? Sure, that requires amajor version bump but it might be worth it given the small adoption the library has

I simply didn't think that this should make it a nextcloud-link v2.0.0… from a semver POV, yes, it could also be done like that, but feels odd.

**Comment by kevinley** [comment]
I could add a deadline when I'll remove the function alltogether

**Comment by floriansimon1** [comment]
> I could add a deadline when I'll remove the function alltogether

Yeah, you could say something along the line of "this will be removed in version 2"

**Comment by kevinley** [comment]
Yess, update incoming

**Pull request closed without merging**